src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java

```diff
@@ -26575,5 +26577,89 @@ private static final void registerOpticalComponentRecipes() {
+        GT_Recipe.GT_Recipe_Map.sFluidHeaterRecipes.addRecipe(
+                true,
+                new ItemStack[] {NI},
+                null,
+                null,
+                new FluidStack[] {Materials.Grade1PurifiedWater.getFluid(1000L)},
+                new FluidStack[] {Materials.Grade2PurifiedWater.getFluid(1000L)},
+                5000,
+                65440,
+                0);
+        GT_Values.RA.addDistillationTowerRecipe(
+                Materials.Grade2PurifiedWater.getFluid(1000L),
+                new FluidStack[] {
+                    Materials.Grade3PurifiedWater.getFluid(1000L),
+                },
+                null,
+                500 * 10,
+                138880);
+        GT_Values.RA.addLaserEngraverRecipe(
+                new ItemStack[] {GT_ModHandler.getModItem("bartworks", "gt.bwMetaGeneratedlens", 0L, 25)
+                }, // Hedenbergite Lens
+                new FluidStack[] {Materials.Grade3PurifiedWater.getFluid(1000L)},
+                new ItemStack[] {NI},
+                new FluidStack[] {Materials.Grade4PurifiedWater.getFluid(1000L)},
+                500 * 10,
+                261760,
+                true);
+        GT_Values.RA.addMultiblockChemicalRecipe(
+                new ItemStack[] {
+                    GT_Utility.copyAmount(0, Materials.Zeolite.getDust(5)), GT_Utility.getIntegratedCircuit(1)
+                },
+                new FluidStack[] {Materials.Grade4PurifiedWater.getFluid(1000)},
+                new FluidStack[] {
+                    Materials.Grade5PurifiedWater.getFluid(1000),
+                },
+                null,
+                500 * 10,
+                523520);
+        GT_Values.RA.addLaserEngraverRecipe(
+                new ItemStack[] {GT_ModHandler.getModItem("bartworks", "gt.bwMetaGeneratedlens", 0L, 43)
+                }, // Hedenbergite Lens
+                new FluidStack[] {Materials.Grade5PurifiedWater.getFluid(1000L)},
+                new ItemStack[] {NI},
+                new FluidStack[] {Materials.Grade6PurifiedWater.getFluid(1000L)},
+                500 * 10,
+                1047040,
+                true);
+        GT_Values.RA.addDistillationTowerRecipe(
+                Materials.Grade6PurifiedWater.getFluid(1000L),
+                new FluidStack[] {
+                    Materials.Grade7PurifiedWater.getFluid(1000L),
+                },
+                null,
+                500 * 10,
+                2094080);
+
+        ;
+    }
+
+    private static final void registerOpticalComponentRecipes() {
+        // Optical Boule
+        GT_Values.RA.addLaserEngraverRecipe(
+                new ItemStack[]{
+                        ItemList.Circuit_Silicon_Ingot5.get(1L),
+                        GT_Utility.copyAmount(0L, GT_ModHandler.getModItem("bartworks", "gt.bwMetaGeneratedlens", 1L, 36)),
+                        GT_Utility.copyAmount(0L, GT_ModHandler.getModItem("bartworks", "gt.bwMetaGeneratedlens", 1L, 23)),
```

Comment:
these two.

Reply:
This was never meant to get here, Actually I don't know where it came from lol.

Reply:
delete?

Reply:
yeah its gone in my code already, I did a bunch of rebasing to get the correct stuff, but it was truly a mess.